market_nextdoor_api/views/vendor_views.py

```diff
@@ -108,4 +108,22 @@ def update_preorder(preorder, data):
 
 def delete_preorder(preorder):
   preorder.delete()
-  return Response(status=status.HTTP_204_NO_CONTENT)
\ No newline at end of file
+  return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+#manytomany views testing
+@api_view(['GET', 'POST'])
+def preorder_test_list(request, vendor_id):
+  try:
+    check_vendor = Vendor.objects.get(pk=vendor_id)
+  except Vendor.DoesNotExist:
+    return Response(status=status.HTTP_404_NOT_FOUND)
+
+  if request.method == 'GET':
+    return get_preorder_test_list(request, check_vendor)
+
+def get_preorder_test_list(request, check_vendor):
+  preorder_tests = Preorder_test.objects.filter(items__vendor=check_vendor).distinct()
```

Comment:
is the .distinct operation here built into django? ORM at all or just a method that comes as a part of library?

Reply:
.distint() is part of the Django Framework and is part of the Django ORM.